ufl/finiteelement.py

```diff
@@ -148,7 +148,7 @@ def components(self) -> _typing.Dict[_typing.Tuple[int, ...], int]:
         offset = 0
         c_offset = 0
         for e in self.sub_elements:
-            for i, j in enumerate(np.ndindex(e.value_shape)):
+            for i, j in enumerate(np.ndindex(e.reference_value_shape)):
```

Comment:
This change shouldn't be in this PR
```suggestion
            for i, j in enumerate(np.ndindex(e.value_shape)):
```

Reply:
I'll revert it once 307 is merged.